assets/js/base/hooks/use-store-cart-item.js

```diff
@@ -22,32 +24,60 @@ import { useStoreCart } from './use-store-cart';
  */
 export const useStoreCartItem = ( cartItemKey ) => {
 	const { cartItems, cartIsLoading } = useStoreCart();
-	const cartItem = cartItems.filter( ( item ) => item.key === cartItemKey );
-
-	const results = useSelect(
-		( select, { dispatch } ) => {
+	const [ cartItem, setCartItem ] = useState( {
+		key: '',
+		isLoading: true,
+		cartData: {},
+		quantity: 0,
+		isPending: false,
+		changeQuantity: () => void null,
+		removeItem: () => void null,
```

Comment:
How will we keep this `cartItem` defaults up to date? This is replicated in a few places now. If we iterate on this schema in future, will we need to manually keep this up to date or will we be guided there by bugs or tooling?

Reply:
We could add a typedef for this, then tooling would warn us about JS code that's out of date with the schema (as long as we update the type!).

Reply:
I think this default schema is actually wrong here because it's mixing `cartItem` with the object returned from the hook `StoreCartItem`. So I'll need to update this.

With regards to the _default_ shape, a typedef would definitely help here but only if `cartItem` is exposed outside of this hook. I'll fix this default and consider typedef in a followup.